### DISCUSSION TITLE
SeleniumBase different work process on Linux

### DISCUSSION THREAD

**Discussion by FindingBen** [seleniumbase/SeleniumBase#3190] [Q&A] [answered]
I asked this question before and I did get an answer but it wasn't really helping me as I think you didn't understood the question.
Basically I have a seleniumbase script on EC2 Amazon Linux system. The script works perfectly on windows locally but when I try it on EC2 it only works until some point and then it seems like its not loging my user properly.

Below in code you can see that I made print statements, when I run script on EC2 they are all being reached which leads me to believe that seleniumbase is initialized and working. But when I reach the password step, I insert password and then my script click next, for some reason the page that is suppose to appear is not appearing. I made a print statement to print the page title and its printing Sign up / Snapchat instead of Ads Manager which should be the case after user logs in.

So the selenium works on linux but the process of inserting the info and then clicking the button seems not to give the correct result. Sorry for a long request but I hope you can help me out as its kinda urgent.

`
def login(self):
        with SB(headless2=True,undetectable=True) as sb:
            print("AS")
            try:
                print('Ok reached here')
                sb.driver.uc_open_with_reconnect("https://accounts.snapchat.com/accounts/v2/login", 4)
                sb.sleep(4)
                print('Hey am here')
                sb.execute_script("""
                        var elements = document.querySelectorAll("section[class*='sdsm-modal sdsm-modal-blocking background--white css-xqeail']");
                        elements.forEach(function(element) {
                            element.parentNode.removeChild(element);
                        });
                """)

                sb.driver.wait_for_element_visible("input[name='accountIdentifier']", timeout=10)

                sb.driver.type("input[name='accountIdentifier']", self.username)
                dd=sb.driver.get_title()
                print("USERNAME",dd)
                sb.driver.sleep(3)
                sb.driver.wait_for_element_visible("//button[contains(text(), 'Next')]", timeout=10)

                sb.driver.sleep(2)  # Give some time for the scrolling to complete
                sb.driver.click("//button[contains(text(), 'Next')]")
                sb.driver.sleep(10)
                sb.driver.execute_script("""
                        var elements = document.querySelectorAll("section[class*='sdsm-modal sdsm-modal-blocking background--white css-xqeail']");
                        elements.forEach(function(element) {
                            element.parentNode.removeChild(element);
                        });
                """)
                sb.driver.wait_for_element_visible("input[name='password']", timeout=10)
                sb.driver.type("input[name='password']", self.password)
                
                
                sb.driver.wait_for_element_visible("//button[contains(text(), 'Next')]", timeout=15)
                
                sb.driver.click("//button[contains(text(), 'Next')]")
                
                print('found 4')
                sb.driver.sleep(20)

                
                sb.driver.get(self.invatation_url)
                sb.driver.sleep(15)
                saapagess = sb.driver.get_title()
                print('AFTER PASSWORD:',saapagess)
                try:
                    close = sb.driver.wait_for_element_visible("//button[span[contains(text(), 'Accept All')]]", timeout=10)
                    close.click()
                    sb.driver.sleep(10)
                    print('CLICKED!')
                except Exception as e:
                    print('Error', e)
                print('almost..')
                ss = sb.driver.get_title()
                print(ss)
                sb.driver.wait_for_element_visible("//button[span[contains(text(), 'Accept')]]", timeout=10)
                sb.driver.click("//button[span[contains(text(), 'Accept')]]")
                sb.driver.sleep(7)
                sb.driver.quit()
                print("Logged in successfully!")
            finally:
                sb.driver.quit()
`

**Comment by mdmintz** [comment, marked as answer]
Your entire question could be simplified down to:
> "I'm running SeleniumBase UC Mode in **EC2** and getting detected."

**EC2** has a known IP-address range, which means websites can easily detect that, and then they know the traffic is coming from a bot. That's why you may have heard the phase: "residential proxy servers" whispered around a lot.
Therefore, even if it works locally, it may not work on your Linux EC2 servers if you're trying to evade bot-detection.
